### PR TITLE
Stricter formatting with eslint as single source

### PR DESCRIPTION
**Pull request by 0x80**
* Use prettier-eslint for formatting.

* Enforce all strings to use backticks, except for imports and jsx.

* Makes vscode prettier settings obsolete.

* Allows prettier-eslint IDE plugin for auto-format on save.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -25,5 +25,17 @@
   "rules": {
     "valid-jsdoc": [0],
     "require-jsdoc": [0],
+    "quotes": ["error", "backtick"],
+    "jsx-quotes": ["error", "prefer-double"],
+    "max-len": ["error", 80],
+    "semi": ["error", "never"],
+    "object-curly-spacing": ["error", "always"],
+    "comma-dangle": ["error", {
```
Comment (KyleAMathews) [inline .eslintrc]: It seems some of these are duplicates of Prettier? semi, object-curly-spacing, comma-dangle, jsx-quotes right?

Comment (KyleAMathews) [inline .eslintrc]: Oh I see, so you disable the rules from prettier and just add them here.

Comment (0x80) [inline .eslintrc]: Yes exactly. The eslint rules become the only source of formatting rules. Which makes perfect sense IMO.

Comment (0x80) [inline .eslintrc]: PS: And if you use an IDE that has a prettier-eslint extension / plugin, I encourage you to try it out with format on save set to true. Makes me smile every time 🙂

**Comment by KyleAMathews** [comment]
Thanks!

**Comment by KyleAMathews** [comment]
Found one thing that needed fixed https://github.com/gatsbyjs/gatsby/commit/abe26fd344e99229e47c896f9818dd0b1b334241

This is what I was mentioning in Discord that webpack chokes on backticks in requires. Which is perhaps a bug since node.js is fine. But because of that, to enforce backticks, we have to use es6 imports which explicitly disallow backticks (I guess since eslint doesn't change them).

**Comment by 0x80** [comment]
@KyleAMathews I assumed requires were fine, because you mentioned import and things seemed to be working. Glad that it all worked out!

**Pull request merged**